sdk/tools/azure-openrewrite/src/main/java/com/azure/openrewrite/AddTryCatchToMethodCallRecipe.java

```diff
@@ -0,0 +1,328 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.openrewrite;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import org.jetbrains.annotations.NotNull;
+
+
+import org.openrewrite.Cursor;
+import org.openrewrite.ExecutionContext;
+import org.openrewrite.NlsRewrite;
+import org.openrewrite.Option;
+import org.openrewrite.Recipe;
+import org.openrewrite.Tree;
+import org.openrewrite.TreeVisitor;
+import org.openrewrite.internal.ListUtils;
+import org.openrewrite.internal.lang.NonNull;
+import org.openrewrite.internal.lang.Nullable;
+import org.openrewrite.java.JavaIsoVisitor;
+import org.openrewrite.java.JavaTemplate;
+import org.openrewrite.java.MethodMatcher;
+
+import lombok.EqualsAndHashCode;
+import lombok.Value;
+import org.openrewrite.java.tree.Expression;
+import org.openrewrite.java.tree.J;
+import org.openrewrite.java.tree.MethodCall;
+import org.openrewrite.java.tree.Statement;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.function.Supplier;
+import java.util.regex.Pattern;
+
+/**
+ * Add try-catch to method recipe places all calls to methods matching the provided method
+ * pattern in a try-catch block based off the template provided.
+ * Recipe will attempt to use the JavaParser first, but will also run on code that cannot be
+ * parsed by Open Rewrites Java parser.
+ * Recipe does not check if the method throws the supplied exception, only that the method is
+ * in a suitable try-catch block.
+ * If the template is not syntactically correct, the recipe will not make any changes.
+ * @author Annabelle Mittendorf Smith
+ */
+
+@Value
+@EqualsAndHashCode(callSuper = false)
+public class AddTryCatchToMethodCallRecipe extends Recipe {
```

Comment:
Maybe I'm misunderstanding this recipe a bit, but is this too generic? Would we be better served by a more restrictive recipe that manages what we know changed from not throwing an exception to throwing an exception? Ex, `BinaryData.toObject` changed from having only runtime exceptions to throwing an `IOException`, should we make this recipe specific for that and similar cases?

Reply:
@alzimmermsft I didn't write these recipes so I'm not entirely sure the intention behind this particular one. @srnagar @samvaity Can either of you offer any insight into this recipe?